docs/how-tos/add-subgraph-schema.md

```diff
@@ -0,0 +1,237 @@
+# Add Subgraph Schema
+
+## Preface
+
+This guide will explain how to introduce or update a the schema of a subgraph via a Pull Request to the [graph-federation](https://github.com/DiamondLightSource/graph-federation/) this repository.
+Code snippets will be given in the form of GitHub actions.
+
+## Generate Subgraph Schema
+
+To begin, we must generate the subgraph schema of our application.
+How this is performed will differ depending on your chosen language and GraphQL library.
+For instance;
+
+- [`strawberry` (Python)](https://strawberry.rocks/) users can use`strawberry export-schema my_package:schema` as described in the [schema export docs].
+- [`async-graphql` (Rust)](https://crates.io/crates/async-graphql) users should create a command which builds and exports the schema as described on the [SDL export page of the book](https://async-graphql.github.io/async-graphql/en/sdl_export.html).
+
+Your CI job should checkout the source, setup your environment and generate the schema, as shown below.
+It is useful to upload the schema as an artifact to allow download both in future workflow jobs and for review.
+
+!!! example "GitHub Actions Schema Generation"
+
+    ```yaml
+    jobs:
+      generate_schema:
+        runs-on: ubuntu-latest
+        steps:
+          - name: Checkout source
+            uses: actions/checkout@v4.1.7
+
+          - name: Setup Python
+            uses: actions/setup-python@v5.2.0
+
+          - name: Install (Dev) Dependencies
+            run: pip install .[dev]
+            
+          - name: Generate Schema
+            run: strawberry export-schema my_app:schema > schema.graphql
+
+          - name: Upload Schema Artifact
+            uses: actions/upload-artifact@v4.4.0
+            with:
+              name: graphql-schema
+              path: schema.graphql
+    ```
+
+## Check Schema Validity
+
+To check our schema is valid we will use the [Apollo Rover CLI](https://www.apollographql.com/docs/rover/).
+We can retrieve the exisitng schema and configuration by cloning the [`graph-federation` repository](https://github.com/DiamondLightSource/graph-federation/) - in which you will find a `supergraph-config.yaml` file describing the federation and a number of schemas in the `schema/` directory.
+
+Once cloned, we can add our subgraph definition to the `schema/` directory and a descriptor to the `supergraph-config.yaml`.
```

Comment:
```suggestion
Once cloned, we can add our subgraph definition to the `schema/` directory and add a descriptor to the `supergraph-config.yaml`.
```

Reply:
Not sure this suggestion is grammatically correct :sweat_smile: